mlem/core/dataset_type.py

```diff
@@ -140,6 +172,33 @@ def get_model(self) -> Type[BaseModel]:
         return create_model("Primitive", __root__=(self.to_type, ...))
 
 
+DATA_FILE = "data.txt"
+
+
+class PrimitiveWriter(DatasetWriter):
+    type: ClassVar[str] = "primitive"
+
+    def write(
+        self, dataset: DatasetType, storage: Storage, path: str
+    ) -> Tuple[DatasetReader, Dict]:
+        with storage.open(path) as (f, art):
+            pickle.dump(dataset.data, f)
+        return PrimitiveReader(dataset_type=dataset), {self.art_name: art}
+
+
+class PrimitiveReader(DatasetReader):
+    type: ClassVar[str] = "primitive"
+
+    def read(self, artifacts: Dict) -> DatasetType:
+        if len(artifacts) != 1:
+            raise ValueError(
+                f"Wrong artifacts {artifacts}: should be one {DATA_FILE} file"
+            )
+        with artifacts[DatasetWriter.art_name].open() as f:
+            data = pickle.load(f)
+            return self.dataset_type.copy().bind(data)
+
+
 class ListTypeWithSpec(DatasetType):
```

Comment:
We don't need this anymore. Remove this one and merge `ListDatasetType` and `SizedTypedListType` into single class. Also please add to the docstrings of `ListDatasetType` and `TupleLikeListType` what is the difference between them